apps/docs/content/guides/auth/migrations/auth0.mdx

```diff
@@ -0,0 +1,208 @@
+
+For more information on these concepts, you can refer to our guide for [Users](/docs/guides/auth/users) and [Identities](/docs/guides/auth/identities).
+
+## Mapping user-specific metadata and custom claims
+
+Supabase Auth provides 2 fields which you can use to map user-specific metadata from Auth0:
+
+- `auth.users.raw_user_meta_data` : For storing non-sensitive user metadata that the user can update (e.g full name, age, favorite color).
+- `auth.users.raw_app_meta_data` : For storing non-sensitive user metadata that the user should not be able to update (e.g pricing plan, access control roles).
+
+Both columns are accessible from the admin user methods. To create a user with custom metadata, you can use the following method:
+
+```ts
+const { data, error } = await supabase.auth.admin.createUser({
+  email: 'foo@example.com',
+  user_metadata: {
+    full_name: 'Foo Bar',
+  },
+  app_metadata: {
+    role: 'admin',
+  },
+})
+```
+
+<Admonition type="warning">
+
+These fields will be exposed in the user's access token JWT so we recommend not to store excessive metadata in these fields.
+
+</Admonition>
+
+These fields are stored as columns in the `auth.users` table using the `jsonb` type. Both fields can be updated by using the admin [`updateUserById` method](/docs/reference/javascript/auth-admin-updateuserbyid). If you want to allow the user to update their own `raw_user_meta_data` , you can use the [`updateUser` method](/docs/reference/javascript/auth-updateuser).
+
+If you have a lot of user-specific metadata to store, it is recommended to create your own table in a private schema that uses the user id as a foreign key. For example:
+
+```sql
+create table private.user_metadata (
+	id int generated always as identity,
+	user_id uuid references auth.users(id) on delete cascade,
+	user_metadata jsonb
+);
+```
+
+## Frequently Asked Questions (FAQ)
+
+> I have IDs assigned to existing users in my database, how can I maintain these IDs?
+
+All users stored in Supabase Auth use the UUID V4 format as the ID. If your UUID format is identical, you can simply specify it in the admin create user method like this:
+
+<Admonition type="note">
+
+New users in Supabase Auth will always be created with a UUID V4 ID by default.
+
+</Admonition>
+
+```ts
+// specify a custom id
+const { data, error } = await supabase.auth.admin.createUser({
+  id: 'e7f5ae65-376e-4d05-a18c-10a91295727a',
+  email: 'foo@example.com',
+})
+```
+
+> How can I allow my users to retain their existing password?
+
+Supabase Auth never stores passwords as plaintext. Since Supabase Auth supports reading BCrypt and argon2 password hashes, you can simply import your users passwords if they use the same hashing algorithm. New users in Supabase Auth who use password-based sign-in methods will always use a BCrypt hash. Passwords are stored in the `auth.users.encrypted_password` column.
```

Comment:
Maybe I am naive here but I'd want to know about salting - is that something BCrypt and argon2 handle under the hood? Or does the user need to config?

Reply:
yup both bcrypt and argon2 handle salting for you